db/db_test.cc

```diff
@@ -5314,41 +5314,40 @@ TEST_F(DBTest, DynamicMiscOptions) {
 #endif  // ROCKSDB_LITE
 
 TEST_F(DBTest, L0L1L2AndUpHitCounter) {
+  const int kNumKeysPerDb = 30000;
+  const int kNumLevels = 3;
+  const int kNumKeysPerLevel = kNumKeysPerDb / kNumLevels;
+
   Options options = CurrentOptions();
-  options.write_buffer_size = 32 * 1024;
-  options.target_file_size_base = 32 * 1024;
-  options.level0_file_num_compaction_trigger = 2;
-  options.level0_slowdown_writes_trigger = 2;
-  options.level0_stop_writes_trigger = 4;
-  options.max_bytes_for_level_base = 64 * 1024;
-  options.max_write_buffer_number = 2;
-  options.max_background_compactions = 8;
-  options.max_background_flushes = 8;
   options.statistics = ROCKSDB_NAMESPACE::CreateDBStatistics();
-  CreateAndReopenWithCF({"mypikachu"}, options);
+  Reopen(options);
 
-  int numkeys = 20000;
-  for (int i = 0; i < numkeys; i++) {
-    ASSERT_OK(Put(1, Key(i), "val"));
+  // Place one file on each of L0, L1, and L2.
```

Comment:
How are you guaranteed this is "one file" on each level?  Or are you suggesting there will be one file after the flush/compact?

Reply:
Yes after the flush/compact below. I rewrote the loops in a clearer way.